dask/sharedict.py

```diff
@@ -0,0 +1,88 @@
+from toolz import concat
+from collections import Mapping
+
+
+class ShareDict(Mapping):
+    """ A MutableMapping composed of other MutableMappings
+
+    This mapping is composed of a mapping of dictionaries
```

Comment:
I think this should mention that all dictionaries should have disjoint keysets, otherwise behaviour is undefined. Also, dictionaries should probably not be mutated after they are incorporated in the ShareDict (this doesn't seem to bother the current implementation, but it could if later tweaks add some caching to improve performance).